File: solutions/py/01-2.py
import math
import sys

def get_fuel(mass):
    fuel = math.floor(mass / 3) - 2
    if fuel <= 0:
        return 0
    return fuel + get_fuel(fuel)

fuels = []
for line in sys.stdin:
    if line.rstrip() == "":
        break
    fuels.append(get_fuel(int(line)))

print(sum(fuels))

File: 19/cpp/01-2.cpp
#include<iostream>

int getFuel(int mass) {
	int fuel = (mass / 3) - 2;
	if (fuel <= 0) return 0;
	return fuel + getFuel(fuel);
}

int main() {
	int mass, sum = 0;
	while (std::cin >> mass) sum += getFuel(mass);
	std::cout << sum << std::endl;
}

